src/Sql/Sql/TransparentDataEncryption/Cmdlet/AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorBase.cs

```diff
@@ -0,0 +1,149 @@
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Adapter;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet
+{
+    public abstract class AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorBase : AzureSqlCmdletBase<IEnumerable<AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel>, AzureSqlDatabaseTransparentDataEncryptionArmAdapter>
+    {
+        /// <summary>
+        /// Parameter sets
+        /// </summary>
+        protected const string DefaultParameterSet = "AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorDefaultParameterSet";
+        protected const string InputObjectParameterSet = "AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorInputObjectParameterSet";
+        protected const string ResourceIdParameterSet = "AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorResourceIdParameterSet";
+
+        /// <summary>
+        /// Gets or sets the instance Object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The instance input object")]
+        [ValidateNotNullOrEmpty]
+        [Alias("InputObject")]
+        public AzureSqlManagedInstanceModel Instance { get; set; }
+
+        /// <summary>
+        /// Gets or sets the instance Resource Id
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The instance resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string InstanceResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The Resource Group Name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the managed instance name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 1,
+            HelpMessage = "The instance name")]
+        [ValidateNotNullOrEmpty]
+        public string InstanceName { get; set; }
```

Comment:
This probably needs to be called Name - if the object this names matches the cmdlet noun (i.e ManagedServiceInstance, then it should be called 'Name' - it is OK to have an alias to this

Reply:
I think Name could be confusing in this context, it could mean ManagedInstanceTransparentDataEncryptionProtector Name or ManagedInstance Name. Hence I think it is important to use InstanceName instead of just Name